charts/opentelemetry-operator/values.yaml

```diff
@@ -186,6 +186,11 @@ manager:
   #   drop:
   #   - ALL
 
+  # Configuration for enabling the `operator.targetallocator.mtls` feature gate.
+  targetAllocator:
+    mtls:
+      enabled: false
```

Comment:
IMO we don't need to nest this so deeply and could probably just add a section for featuregates to enable and disable more explicitly.

Reply:
I'm thinking about how this might work. Currently, feature gates are explicitly defined using a string value. I'd likely want to convert the string to an object that we can use to define feature gates. Long-term, that seems to be the best idea as it allows the chart to handle feature gates like this one that require extra configuration.

That does mean that we're going to need to bump the chart version though. If we're following semver strictly, we'd need to bump the major version of the chart.

Was that what you had in mind?

Reply:
i'm talking specifically about what's exposed to the user i.e.
```yaml
featureGates:
  targetAllocatorMtls:
    enabled: false # default
```

Where featuregates is a map with no required variables. I'm not sure why we'd need to bump the major version of the chart for that?

Reply:
I can definitely do this, but I'm talking about bumping the major version of the chart because it is a breaking change to the values file. The current `featureGates` value takes a simple string. We'd be changing that string to an object. For people who don't set the featureGates, it'd be fine but for people do, they'd need to use the new way.

Reply:
ahh i see what you mean, in that case we could just do a string contains with a helper template? Or we could just do like `featureGateMap` and deprecate the other method?

Reply:
sounds good to me!

Reply:
Updated. The expected behavior is that if the `featureGatesMap` property is not empty, the featureGates string will be ignored. No one should be using that value currently so by default, updating the chart should do nothing for those people. If they decide to move to the featureGatesMap, it should take precedence as it is designed to ensure all other dependencies are met.

How does a value deprecation get signaled?

Reply:
just via a comment usually. you can also include it in the upgrade.md file :)

Reply:
I updated both the comment and added a note to the UPGRADING.md file. I think this PR is ready for another review.